packages/block-editor/src/store/utils.js

```diff
@@ -1,35 +1,52 @@
+/**
+ * External dependencies
+ */
+import createSelector from 'rememo';
+
 /**
  * Internal dependencies
  */
 import { INSERTER_PATTERN_TYPES } from '../components/inserter/block-patterns-tab/utils';
 
-const EMPTY_ARRAY = [];
+export const getUserPatterns = createSelector(
+	( state ) => {
+		const userPatterns = state.settings.__experimentalReusableBlocks ?? [];
+		const userPatternCategories =
+			state.settings.__experimentalUserPatternCategories ?? [];
```

Comment:
These two arrays are never returned from the selector, so the static constant `EMPTY_ARRAY` is not really needed.

Reply:
Could be easy to accidentally slip in later though.

Reply:
Although we're anyway mapping below I see now.